test/metabase/driver/postgres/test_data.clj

```diff
@@ -0,0 +1,257 @@
+(ns metabase.driver.postgres.test-data
+  (:require
+   [clojure.java.io :as io]
+   [clojure.java.jdbc :as jdbc]
+   [clojure.string :as str]
+   [java-time.api :as t]
+   [metabase.test.data.sql :as sql.tx]
+   [metabase.test.data.sql-jdbc.spec :as spec]
+   [metabase.test.test-data :as test.test-data]
+   [metabase.util :as u]
+   [metabase.util.date-2 :as u.date]
+   [metabase.util.log :as log]
+   [metabase.util.malli :as mu]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private database-sql
+  (str/join
+   \newline
+   ["DO $$ BEGIN"
+    "  PERFORM pg_terminate_backend(pg_stat_activity.pid)"
+    "  FROM pg_stat_activity"
+    "  WHERE pid <> pg_backend_pid()"
+    "    AND pg_stat_activity.datname = 'test-data';"
+    "END $$;"
+    ""
+    "-----"
+    ""
+    "DROP DATABASE IF EXISTS \"test-data\";"
+    ""
+    "-----"
+    ""
+    "CREATE DATABASE \"test-data\";"]))
+
+#_{:clj-kondo/ignore [:discouraged-var]}
+(mu/defn ^:private print-field-sql
+  [driver :- ::test.test-data/driver
+   field  :- ::test.test-data/schema.field]
+  ;; write field name
+  (printf "\"%s\"" (:name field))
+  ;; write field type
+  (print " ")
+  (print (u/lower-case-en (sql.tx/field-base-type->sql-type driver (:base-type field))))
+  ;; write field modifiers
+  (when (:pk? field)
+    (print " PRIMARY KEY"))
+  (when (:not-null? field)
+    (print " NOT NULL"))
+  (when (:unique? field)
+    (print " UNIQUE"))
+  (when-let [[table-name field-name] (:fk field)]
+    (printf " REFERENCES \"%s\" (\"%s\")" table-name field-name))
+  (when-let [comment (:comment field)]
+    (printf " COMMENT '%s'" (str/replace comment #"'" "''"))))
```

Comment:
The code ends up being pretty imperative (just a series of calls to `print` and friends or other functions that call `print`) but this is a easy and natural way to spit out SQL statements like these. Definitely a lot easier to follow than the nonsense we have in the current code

https://github.com/metabase/metabase/blob/856b449626db3d3e07dc216bfdc369fd90cf747b/test/metabase/test/data/sql/ddl.clj#L60-L86

(which ends up being fairly imperative anyway, but uses an atom to collect the output).